src/source/Signaling/Signaling.c

```diff
@@ -373,11 +389,19 @@ STATUS validateSignalingClientInfo(PSignalingClient pSignalingClient, PSignaling
     STATUS retStatus = STATUS_SUCCESS;
 
     CHK(pSignalingClient != NULL && pClientInfo != NULL, STATUS_NULL_ARG);
-    CHK(pClientInfo->version <= SIGNALING_CLIENT_INFO_CURRENT_VERSION, STATUS_SIGNALING_INVALID_CLIENT_INFO_VERSION);
+    CHK((pClientInfo->version & (~SIGNALING_CLIENT_INFO_INTERNAL_SENTINEL_BIT)) <= SIGNALING_CLIENT_INFO_CURRENT_VERSION, STATUS_SIGNALING_INVALID_CLIENT_INFO_VERSION);
     CHK(STRNLEN(pClientInfo->clientId, MAX_SIGNALING_CLIENT_ID_LEN + 1) <= MAX_SIGNALING_CLIENT_ID_LEN, STATUS_SIGNALING_INVALID_CLIENT_INFO_CLIENT_LENGTH);
 
-    // Store and validate
-    pSignalingClient->clientInfo = *pClientInfo;
+    // Store and validate.
+    // NOTE: We will check whether an internal structure has been passed in for testing purposes by
+    // checking the sentinel bit is set.
+    if ((pClientInfo->version & SIGNALING_CLIENT_INFO_INTERNAL_SENTINEL_BIT) != 0) {
```

Comment:
It's not very obvious what SIGNALING_CLIENT_INFO_INTERNAL_SENTINEL_BIT is supposed to, and it feels hacky to overload version with this extra functionality. Cant we just add a boolean?

Reply:
The issue is that we are trying to not expose this to the customers code. Yes it's kind of hacky but this is supposed to be used internally and for test injection purposes only

Reply:
what about adding a boolean and a compile time flag to check against the boolean? The issue with your approach is that it is gonna set a style here and once it spreads we will have two versions of version which just adds more unnecessary complexity.

Reply:
The main idea here is to NOT expose any of this in the public headers.